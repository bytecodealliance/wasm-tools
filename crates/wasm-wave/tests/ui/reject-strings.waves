// Reject surrogates.
string("\u{d800}");
string("\u{dbff}");
string("\u{dc00}");
string("\u{dcff}");
string("\u{d800}\\u{dc00}");
// Reject invalid values.
string("\u{110000}");
string("\u{ffffffff}");
string("\u{80000000}");
// Reject invalid syntax.
string("\u{-1}");
string("\u{+1}");
// Missing mandatory escapes
string(""");
string("\");
string("
");
// Multiline missing mandatory line breaks
string("""""");
string(""" """);
// Multiline invalid delimiters
string(""""
""");
string("""
"");
string("""
"""");
string("""extra
""");
string("""
extra""");
// Multiline double quote triplet in content
string("""
  """
  """);
string("""
  Between """ words
  """);
// Multiline invalid escapes
string("""
Invalid surrogate: \u{d800}
""");
string("""
Invalid escape: \j
""");