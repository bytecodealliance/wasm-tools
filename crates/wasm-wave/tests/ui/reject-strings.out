// Reject surrogates.
invalid character escape at 8..9
invalid character escape at 8..9
invalid character escape at 8..9
invalid character escape at 8..9
invalid character escape at 8..9
// Reject invalid values.
invalid character escape at 8..9
invalid character escape at 8..9
invalid character escape at 8..9
// Reject invalid syntax.
invalid token at 7..8
invalid token at 7..8
// Missing mandatory escapes
invalid token at 7..10
invalid token at 7..11
invalid token at 7..8
// Multiline missing mandatory line breaks
invalid multiline string: opening """ must be followed immediately by newline at 10..11
invalid multiline string: opening """ must be followed immediately by newline at 10..11
// Multiline invalid delimiters
invalid multiline string: opening """ must be followed immediately by newline at 10..11
invalid token at 7..10
invalid token at 14..16
invalid multiline string: opening """ must be followed immediately by newline at 10..11
invalid multiline string: closing """ must be on a line preceded only by spaces at 13..16
// Multiline double quote triplet in content
invalid token at 19..22
unexpected token: LabelOrKeyword at 25..30
// Multiline invalid escapes
invalid character escape at 30..31
invalid character escape at 27..28
